nilearn/input_data/nifti_maps_masker.py

```diff
@@ -184,12 +184,12 @@ def fit(self, X=None, y=None):
                    _utils._repr_niimgs(self.maps_img,
                                        shorten=(not self.verbose)),
                    verbose=self.verbose)
-
-        self.maps_img_ = _utils.check_niimg_4d(self.maps_img, dtype=self.dtype)
+        self.maps_img_ = _utils.check_niimg(self.maps_img, dtype=self.dtype)
+        self.maps_img_ = np.expand_dims(self.maps_img_, 0) if len(
```

Comment:
I don't think this will work for several reasons:

- `np.exand_dims` expects an array as [input](https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html) and here it can receive a string. 
- `np.exand_dims` also returns an array, which will cause `clean_img` to raise a TypeError:

```python
import numpy as np
from nilearn._utils import testing, as_ndarray, data_gen
from nilearn.image import clean_img, get_data
affine = np.diag((4, 4, 4, 1))
n_regions = 3
shape3 = (2, 2, 2)

labels33_img = data_gen.generate_labeled_regions(shape3, n_regions,
                                                     affine=affine)
clean_img(np.expand_dims(get_data(labels33_img), 0))
```
```
TypeError                                 Traceback (most recent call last)
<ipython-input-6-5f193dc4c2fc> in <module>
      8 labels33_img = data_gen.generate_labeled_regions(shape3, n_regions,
      9                                                      affine=affine)
---> 10 clean_img(np.expand_dims(get_data(labels33_img), 0))
```

- When calling `clean_img`, the signals in the last dimension are filtered, so I think you need to add a dimension "at the end" rather than "at the beginning".

Unless I'm missing something, I think `self.maps_img_ = _utils.check_niimg(self.maps_img, dtype=self.dtype, atleast_4d=True)` should do the trick here.

Reply:
Hello @NicolasGensollen , Thank you for your review.
In fact when i was exploring the check_niimg function i discoverd the fact that it is loading the image and returning the loaded image in the case of a filename
    niimg = load_niimg(niimg, dtype=dtype)
    if ensure_ndim == 3 and len(niimg.shape) == 4 and niimg.shape[3] == 1:
        # "squeeze" the image.
        data = _safe_get_data(niimg)
        affine = niimg.affine
        niimg = new_img_like(niimg, data[:, :, :, 0], affine)
    if atleast_4d and len(niimg.shape) == 3:
        data = _get_data(niimg).view()
        data.shape = data.shape + (1, )
        niimg = new_img_like(niimg, data, niimg.affine)

    if ensure_ndim is not None and len(niimg.shape) != ensure_ndim:
        raise DimensionError(len(niimg.shape), ensure_ndim)

    if return_iterator:
        return (_index_img(niimg, i) for i in range(niimg.shape[3]))

    return niimg

Now, for the clean_img function, according to the parameters it can handle only 4d images, this is why in my last commit i added a condiitional operator on the length of the shape of the image (even loaded in the filename case). 
Now, in my new commit, i tested the case with at_least4d=True and it worked with all the tests (This may due to the fact that check_niimg with this parameter is adjusting the dimensionality of the image to 4d). 
Therefore, i'm comitting the changes now